otx/algorithms/anomaly/tasks/nncf.py

```diff
@@ -144,9 +144,17 @@ def load_model(self, otx_model: Optional[ModelEntity]) -> AnomalyModule:
                     pl_modules[key] = model_data["model"][key]
             model_data["model"] = nncf_modules
 
+            dataloader = None
+            if hasattr(self, "trainer") and hasattr(self.trainer, "datamodule"):
+                if self.trainer.datamodule.train_dataset is not None:
+                    dataloader = self.trainer.datamodule.train_dataloader()
+                elif self.trainer.datamodule.test_dataset is not None:
+                    dataloader = self.trainer.datamodule.test_dataloader()
+
             self.compression_ctrl, model.model = wrap_nncf_model(
                 model.model,
                 self.optimization_config["nncf_config"],
+                dataloader=dataloader,  # type:ignore
```

Comment:
Is it okay dataloader is None? If it is okay then what case could be?

Reply:
If I understand correctly, dataloader will always be None in this case as `wrap_nncf_model` requires either a trained nncf model or a dataloader. In this case we are loading weights from a trained model into nncf. But @AlexanderDokuchaev can clarify this better as he is the owner of `wrap_nncf_model` code